dowhy/causal_model.py

```diff
@@ -18,347 +19,378 @@
-        :param method_name: name of the refutation method
-        :param kwargs:  (optional) additional arguments that are passed directly to the refutation method. Can specify a random seed here to ensure reproducible results ('random_seed' parameter). For method-specific parameters, consult the documentation for the specific method. All refutation methods are in the causal_refuters subpackage.
-
-        :returns: an instance of the RefuteResult class
-
-        """
-        if method_name is None:
-            pass
-        else:
-            refuter_class = causal_refuters.get_class_object(method_name)
-
-        refuter = refuter_class(
-            self._data,
-            identified_estimand=estimand,
-            estimate=estimate,
-            **kwargs
-        )
-        res = refuter.refute_estimate()
-        return res
-
-    def view_model(self, layout="dot"):
-        """View the causal DAG.
-
-        :param layout: string specifying the layout of the graph.
-
-        :returns: a visualization of the graph
-
-        """
-        self._graph.view_graph(layout)
-
-    def interpret(self, method_name=None, **kwargs):
-        """Interpret the causal model.
-
-        :param method_name: method used for interpreting the model. If None,
-                            then default interpreter is chosen that describes the model summary and shows the associated causal graph.
-        :param kwargs:: Optional parameters that are directly passed to the interpreter method.
-
-        :returns: None
-
-        """
-        if method_name is None:
-            self.summary(print_to_stdout=True)
-            self.view_model()
-            return
-
-        method_name_arr = parse_state(method_name)
-        import dowhy.interpreters as interpreters
-        for method in method_name_arr:
-            interpreter = interpreters.get_class_object(method)
-            interpreter(self, **kwargs).interpret()
-
-    def summary(self, print_to_stdout=False):
-        """Print a text summary of the model.
-
-        :returns: a string containining the summary
-
-        """
-        summary_text = "Model to find the causal effect of treatment {0} on outcome {1}".format(self._treatment, self._outcome)
-        self.logger.info(summary_text)
-        if print_to_stdout:
-            print(summary_text)
-        return summary_text
+	"""Main class for storing the causal model state.
+
+	"""
+
+	def __init__(self, data, treatment, outcome, graph=None,
+				 common_causes=None, instruments=None,
+				 effect_modifiers=None,
+				 estimand_type="nonparametric-ate",
+				 proceed_when_unidentifiable=False,
+				 missing_nodes_as_confounders=False,
+				 **kwargs):
+		"""Initialize data and create a causal graph instance.
+
+		Assigns treatment and outcome variables.
+		Also checks and finds the common causes and instruments for treatment
+		and outcome.
+
+		At least one of graph, common_causes or instruments must be provided.
```

Comment:
you need to change this documentation. Saying that: if none of these variables are provided, then you can use learn_graph later.